x-pack/plugins/synthetics/server/routes/synthetics_service/assets_check.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SyntheticsServerSetup } from '../../types';
+import { SyntheticsRestApiRouteFactory } from '../types';
+import { SYNTHETICS_API_URLS } from '../../../common/constants';
+
+const EXPECTED_PIPELINES = [
+  'synthetics-tcp',
+  'synthetics-http',
+  'synthetics-icmp',
+  'synthetics-browser',
+  'synthetics-browser.screenshot',
+  'synthetics-browser.network',
+];
+
+const EXPECTED_INDEX_TEMPLATES = ['synthetics', ...EXPECTED_PIPELINES];
+
+export const getSyntheticsAssetsCheckRoute: SyntheticsRestApiRouteFactory = () => ({
+  method: 'PUT',
+  path: SYNTHETICS_API_URLS.SYNTHETICS_ASSETS_CHECK,
+  validate: {},
+  writeAccess: true,
+  handler: async ({
+    server,
+  }): Promise<{ hasAllAssets: boolean; reinstalled?: boolean; error?: string }> => {
+    return await verifySyntheticsAssetsChecks(server);
+  },
+});
+
+export const verifySyntheticsAssetsChecks = async (server: SyntheticsServerSetup) => {
+  const esClient = server.coreStart.elasticsearch.client.asInternalUser;
+
+  const [pipelines, indexTemplates] = await Promise.all([
+    esClient.ingest.getPipeline({
+      id: 'synthetics-*',
+    }),
+    esClient.indices.getIndexTemplate({
+      name: 'synthetics*',
+    }),
+  ]);
+  const pipelineIds = Object.keys(pipelines);
+  const missingPipelines = EXPECTED_PIPELINES.filter(
+    (pipeline) => !pipelineIds.some((id) => id.startsWith(pipeline))
+  );
+
+  const indexTemplateNames = indexTemplates.index_templates ?? [];
+  // all index templates should be present
+  const missingIndexTemplates = EXPECTED_INDEX_TEMPLATES.filter(
+    (template) => !indexTemplateNames.some(({ name }) => name === template)
```

Comment:
The `some` issue I talked about above doesn't seem to happen here because of the equality check, in cases where I deleted an overloaded name we still re-installed all the assets, including the missing pipelines I talked about above.